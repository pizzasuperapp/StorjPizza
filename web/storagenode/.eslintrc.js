// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

module.exports = {
    root: true,
    env: {
        node: true
    },
    extends: [
        'plugin:vue/recommended',
        'eslint:recommended',
        '@vue/typescript/recommended',
    ],
    parserOptions: {
        ecmaVersion: 2020
    },
    plugins: ["storj"],
    rules: {
        "linebreak-style": ["error", "unix"],

        'no-console': process.env.NODE_ENV === 'production' ? 'warn' : 'off',
        'no-debugger': process.env.NODE_ENV === 'production' ? 'warn' : 'off',

        "indent": ["warn", 4],
        "vue/html-indent": ["warn", 4],

        "@typescript-eslint/no-unused-vars": [
            "warn", {
                "vars": "all",
                "args": "all",
                "argsIgnorePattern": "^_"
            }],

        '@typescript-eslint/no-empty-function': "off",
        '@typescript-eslint/no-var-requires': "off",

        "vue/multi-word-component-names": ["off"],
        "vue/max-attributes-per-line": ["off"],
        "vue/singleline-html-element-content-newline": ["off"],

        "vue/block-lang": ["error", {"script": {"lang": "ts"}}],
        "vue/html-button-has-type": ["error"],
        "vue/no-unused-properties": ["warn"],
        "vue/no-unused-refs": ["warn"],
        "vue/no-useless-v-bind": ["warn"],

        'vue/no-unregistered-components': ['warn', { ignorePatterns: ['router-link', 'router-view'] }],

        'storj/vue/require-annotation': 'warn',
    },
}