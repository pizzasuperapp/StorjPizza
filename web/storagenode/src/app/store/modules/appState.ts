// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export const APPSTATE_MUTATIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART,',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK: 'SET_DARK',
    SET_NO_PAYOUT_INFO: 'SET_NO_PAYOUT_INFO',
    SET_LOADING_STATE: 'SET_LOADING_STATE',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

export const APPSTATE_ACTIONS = {
    TOGGLE_SATELLITE_SELECTION: 'TOGGLE_SATELLITE_SELECTION',
    TOGGLE_BANDWIDTH_CHART: 'TOGGLE_BANDWIDTH_CHART',
    TOGGLE_EGRESS_CHART: 'TOGGLE_EGRESS_CHART',
    TOGGLE_INGRESS_CHART: 'TOGGLE_INGRESS_CHART',
    CLOSE_ADDITIONAL_CHARTS: 'CLOSE_ADDITIONAL_CHARTS',
    TOGGLE_PAYOUT_CALENDAR: 'TOGGLE_PAYOUT_CALENDAR',
    CLOSE_ALL_POPUPS: 'CLOSE_ALL_POPUPS',
    SET_DARK_MODE: 'SET_DARK_MODE',
    SET_NO_PAYOUT_DATA: 'SET_NO_PAYOUT_DATA',
    SET_LOADING: 'SET_LOADING',
    TOGGLE_PAYOUT_HISTORY_CALENDAR: 'TOGGLE_PAYOUT_HISTORY_CALENDAR',
};

const {
    TOGGLE_SATELLITE_SELECTION,
    TOGGLE_BANDWIDTH_CHART,
    TOGGLE_EGRESS_CHART,
    TOGGLE_INGRESS_CHART,
    CLOSE_ADDITIONAL_CHARTS,
    CLOSE_ALL_POPUPS,
    TOGGLE_PAYOUT_CALENDAR,
} = APPSTATE_MUTATIONS;

class AppState {
    public constructor (
        public isSatelliteSelectionShown = false,
        public isBandwidthChartShown = true,
        public isEgressChartShown = false,
        public isIngressChartShown = false,
        public isPayoutCalendarShown = false,
        public isDarkMode = false,
        public isNoPayoutData = false,
        public isLoading = true,
        public isPayoutHistoryCalendarShown = false,
    ) {}
}

interface AppContext {
    state: AppState;
    commit: (string, ...unknown) => void;
}

export const appStateModule = {
    state: new AppState(),
    mutations: {
        [TOGGLE_SATELLITE_SELECTION](state: AppState): void {
            state.isSatelliteSelectionShown = !state.isSatelliteSelectionShown;
        },
        [TOGGLE_BANDWIDTH_CHART](state: AppState): void {
            state.isBandwidthChartShown = !state.isBandwidthChartShown;
        },
        [TOGGLE_EGRESS_CHART](state: AppState): void {
            state.isEgressChartShown = !state.isEgressChartShown;
        },
        [TOGGLE_INGRESS_CHART](state: AppState): void {
            state.isIngressChartShown = !state.isIngressChartShown;
        },
        [TOGGLE_PAYOUT_CALENDAR](state: AppState, value: boolean): void {
            state.isPayoutCalendarShown = value;
        },
        [CLOSE_ADDITIONAL_CHARTS](state: AppState): void {
            state.isBandwidthChartShown = true;
            state.isIngressChartShown = false;
            state.isEgressChartShown = false;
        },
        [APPSTATE_MUTATIONS.SET_DARK](state: AppState, value: boolean): void {
            state.isDarkMode = value;
        },
        [APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO](state: AppState, value: boolean): void {
            state.isNoPayoutData = value;
        },
        [APPSTATE_MUTATIONS.SET_LOADING_STATE](state: AppState, value: boolean): void {
            state.isLoading = value;
        },
        [CLOSE_ALL_POPUPS](state: AppState): void {
            state.isSatelliteSelectionShown = false;
        },
        [APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR](state: AppState, value: boolean): void {
            state.isPayoutHistoryCalendarShown = value;
        },
    },
    actions: {
        [APPSTATE_ACTIONS.TOGGLE_SATELLITE_SELECTION]: function ({commit, state}: AppContext): void {
            if (!state.isSatelliteSelectionShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_SATELLITE_SELECTION);

                return;
            }

            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_CALENDAR, value);
        },
        [APPSTATE_ACTIONS.TOGGLE_EGRESS_CHART]: function ({commit, state}: AppContext): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);
        },
        [APPSTATE_ACTIONS.TOGGLE_INGRESS_CHART]: function ({commit, state}: AppContext): void {
            if (!state.isBandwidthChartShown) {
                commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
                commit(APPSTATE_MUTATIONS.TOGGLE_EGRESS_CHART);

                return;
            }

            commit(APPSTATE_MUTATIONS.TOGGLE_BANDWIDTH_CHART);
            commit(APPSTATE_MUTATIONS.TOGGLE_INGRESS_CHART);
        },
        [APPSTATE_ACTIONS.SET_DARK_MODE]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_DARK, value);
        },
        [APPSTATE_ACTIONS.SET_NO_PAYOUT_DATA]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.SET_NO_PAYOUT_INFO, value);
        },
        [APPSTATE_ACTIONS.SET_LOADING]: function ({commit}: AppContext, value: boolean): void {
            value ? commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value) :
                setTimeout(() => { commit(APPSTATE_MUTATIONS.SET_LOADING_STATE, value); }, 1000);
        },
        [APPSTATE_ACTIONS.CLOSE_ADDITIONAL_CHARTS]: function ({commit}: AppContext): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ADDITIONAL_CHARTS);
        },
        [APPSTATE_ACTIONS.CLOSE_ALL_POPUPS]: function ({commit}: AppContext): void {
            commit(APPSTATE_MUTATIONS.CLOSE_ALL_POPUPS);
        },
        [APPSTATE_ACTIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR]: function ({commit}: AppContext, value: boolean): void {
            commit(APPSTATE_MUTATIONS.TOGGLE_PAYOUT_HISTORY_CALENDAR, value);
        },
    },
};
