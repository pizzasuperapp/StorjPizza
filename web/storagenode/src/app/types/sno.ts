// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    BandwidthUsed,
    EgressUsed,
    IngressUsed,
    Node,
    SatelliteInfo,
    SatelliteScores,
    Stamp,
    Utilization,
} from '@/storagenode/sno/sno';

/**
 * Holds all node module state.
 */
export class StorageNodeState {
    public info: Node = new Node();
    public utilization: Utilization = new Utilization();
    public satellites: SatelliteInfo[] = [];
    public disqualifiedSatellites: SatelliteInfo[] = [];
    public suspendedSatellites: SatelliteInfo[] = [];
    public selectedSatellite: SatelliteInfo = new SatelliteInfo();
    public bandwidthChartData: BandwidthUsed[] = [];
    public egressChartData: EgressUsed[] = [];
    public ingressChartData: IngressUsed[] = [];
    public storageChartData: Stamp[] = [];
    public storageSummary = 0;
    public bandwidthSummary = 0;
    public egressSummary = 0;
    public ingressSummary = 0;
    public satellitesScores: SatelliteScores[] = [];
    public audits: SatelliteScores = new SatelliteScores();
}
