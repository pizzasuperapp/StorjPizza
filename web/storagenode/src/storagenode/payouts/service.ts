// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import {
    EstimatedPayout,
    PaymentInfoParameters,
    PayoutApi,
    PayoutPeriod,
    Paystub,
    SatelliteHeldHistory,
    SatellitePayoutForPeriod,
    TotalPayments,
    TotalPaystubForPeriod,
} from '@/storagenode/payouts/payouts';

/**
 * PayoutService is used to store and handle node paystub information.
 * PayoutService exposes a business logic related to payouts.
 */
export class PayoutService {
    private readonly payouts: PayoutApi;

    public constructor(api: PayoutApi) {
        this.payouts = api;
    }

    /**
     * Gets summary of paystubs for given period.
     * @param start period start
     * @param end period end
     * @param satelliteId
     */
    public async paystubSummaryForPeriod(start: PayoutPeriod | null, end: PayoutPeriod, satelliteId?: string): Promise<TotalPaystubForPeriod> {
        const paystubs: Paystub[] = await this.payouts.getPaystubsForPeriod(new PaymentInfoParameters(start, end, satelliteId));

        return new TotalPaystubForPeriod(paystubs);
    }

    /**
     * Gets held and paid summary for given period.
     * @param start period start
     * @param end period end
     * @param satelliteId
     */
    public async totalPayments(start: PayoutPeriod, end: PayoutPeriod, satelliteId: string): Promise<TotalPayments> {
        const paystubs: Paystub[] = await this.payouts.getPaystubsForPeriod(new PaymentInfoParameters(start, end, satelliteId));

        return new TotalPayments(paystubs);
    }

    /**
     * Gets list of payout periods that have paystubs for selected satellite.
     * If satelliteId is not provided returns periods for all satellites.
     * @param satelliteId
     */
    public async availablePeriods(satelliteId: string): Promise<PayoutPeriod[]> {
        return await this.payouts.getPayoutPeriods(satelliteId);
    }

    /**
     * Gets list of payout history items for given period by satellites.
     * @param payoutHistoryPeriod year and month representation
     */
    public async payoutHistory(payoutHistoryPeriod: string): Promise<SatellitePayoutForPeriod[]> {
        return await this.payouts.getPayoutHistory(payoutHistoryPeriod);
    }

    /**
     * Gets list of held history for all satellites.
     */
    public async allSatellitesHeldHistory(): Promise<SatelliteHeldHistory[]> {
        return await this.payouts.getHeldHistory();
    }

    /**
     * Gets estimated payout when no data in paystub.
     * @param satelliteId
     */
    public async estimatedPayout(satelliteId: string): Promise<EstimatedPayout> {
        return await this.payouts.getEstimatedPayout(satelliteId);
    }
}
