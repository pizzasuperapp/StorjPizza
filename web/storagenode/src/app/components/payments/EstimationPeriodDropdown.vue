// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <button
        name="Show Period Date Picker"
        class="period-container"
        type="button"
        :class="{ disabled: false }"
        @click.stop="openPeriodDropdown"
    >
        <p class="period-container__label long-text">Custom Date Range</p>
        <p class="period-container__label short-text">Custom Range</p>
        <BlackArrowHide v-if="isCalendarShown" />
        <BlackArrowExpand v-else />
        <PayoutPeriodCalendar
            v-if="isCalendarShown"
            v-click-outside="closePeriodDropdown"
            class="period-container__calendar"
        />
    </button>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import PayoutPeriodCalendar from '@/app/components/payments/PayoutPeriodCalendar.vue';

import BlackArrowExpand from '@/../static/images/BlackArrowExpand.svg';
import BlackArrowHide from '@/../static/images/BlackArrowHide.svg';

import { APPSTATE_ACTIONS } from '@/app/store/modules/appState';

// @vue/component
@Component({
    components: {
        PayoutPeriodCalendar,
        BlackArrowExpand,
        BlackArrowHide,
    },
})
export default class EstimationPeriodDropdown extends Vue {
    /**
     * Indicates if period selection calendar should appear.
     */
    public get isCalendarShown(): boolean {
        return this.$store.state.appStateModule.isPayoutCalendarShown;
    }

    /**
     * Indicates if period selection calendar should be disabled.
     */
    public get isCalendarDisabled(): boolean {
        const nodeStartedAt = this.$store.state.node.selectedSatellite.joinDate;
        const now = new Date();

        return nodeStartedAt.getUTCMonth() === now.getUTCMonth() && nodeStartedAt.getUTCFullYear() === now.getUTCFullYear();
    }

    /**
     * Opens payout period selection dropdown.
     */
    public openPeriodDropdown(): void {
        if (this.isCalendarDisabled) {
            return;
        }

        this.$store.dispatch(APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR, true);
    }

    /**
     * Closes payout period selection dropdown.
     */
    public closePeriodDropdown(): void {
        this.$store.dispatch(APPSTATE_ACTIONS.TOGGLE_PAYOUT_CALENDAR, false);
    }
}
</script>

<style scoped lang="scss">
    .period-container {
        position: relative;
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: center;
        background-color: transparent;
        cursor: pointer;

        &__label {
            margin-right: 8px;
            font-family: 'font_regular', sans-serif;
            font-weight: 500;
            font-size: 16px;
            color: var(--regular-text-color);
        }

        &__calendar {
            position: absolute;
            top: 30px;
            right: 0;
        }
    }

    .active {

        .period-container__label {
            color: var(--navigation-link-color);
        }
    }

    .arrow ::v-deep path {
        fill: var(--period-selection-arrow-color);
    }

    .short-text {
        display: none;
    }

    .disabled {

        .period-container {

            &__label {
                color: #909bad;
            }
        }

        .arrow ::v-deep path {
            fill: #909bad !important;
        }
    }

    @media screen and (max-width: 505px) {

        .period-container__label {
            margin-right: 4px;
        }

        .short-text {
            display: inline-block;
            font-size: 14px;
        }

        .long-text {
            display: none;
        }
    }
</style>
