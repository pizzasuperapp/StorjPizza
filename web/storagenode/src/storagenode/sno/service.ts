// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { StorageNodeApi } from '@/storagenode/api/storagenode';
import { Dashboard, Satellite, Satellites } from '@/storagenode/sno/sno';

/**
 * SNOService is used to store and handle node information.
 * SNOService exposes a business logic related to node.
 */
export class StorageNodeService {
    private readonly node: StorageNodeApi;

    public constructor(api: StorageNodeApi) {
        this.node = api;
    }

    /**
     * Gets dashboard data from server.
     */
    public async dashboard(): Promise<Dashboard> {
        return await this.node.dashboard();
    }

    /**
     * Gets satellite data from server.
     * @param id - satellite id
     */
    public async satellite(id: string): Promise<Satellite> {
        return await this.node.satellite(id);
    }

    /**
     * Gets all satellites data from server.
     */
    public async satellites(): Promise<Satellites> {
        return await this.node.satellites();
    }
}
