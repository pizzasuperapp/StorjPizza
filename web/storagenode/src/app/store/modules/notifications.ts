// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { StoreModule } from '@/app/store';
import { NotificationsState, UINotification } from '@/app/types/notifications';
import { NotificationsService } from '@/storagenode/notifications/service';

export const NOTIFICATIONS_MUTATIONS = {
    SET_NOTIFICATIONS: 'SET_NOTIFICATIONS',
    MARK_AS_READ: 'MARK_AS_READ',
    READ_ALL: 'READ_ALL',
    SET_LATEST: 'SET_LATEST',
};

export const NOTIFICATIONS_ACTIONS = {
    GET_NOTIFICATIONS: 'getNotifications',
    MARK_AS_READ: 'markAsRead',
    READ_ALL: 'readAll',
};

interface NotificationsContext {
    state: NotificationsState;
    commit: (string, ...unknown) => void;
}

/**
 * creates notifications module with all dependencies
 *
 * @param service - payments service
 */
export function newNotificationsModule(service: NotificationsService): StoreModule<NotificationsState> {
    return {
        state: new NotificationsState(),
        mutations: {
            [NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS](state: NotificationsState, notificationsState: NotificationsState): void {
                state.notifications = notificationsState.notifications;
                state.pageCount = notificationsState.pageCount;
                state.unreadCount = notificationsState.unreadCount;
            },
            [NOTIFICATIONS_MUTATIONS.SET_LATEST](state: NotificationsState, notificationsState: NotificationsState): void {
                state.latestNotifications = notificationsState.notifications;
            },
            [NOTIFICATIONS_MUTATIONS.MARK_AS_READ](state: NotificationsState, id: string): void {
                state.notifications = state.notifications.map((notification: UINotification) => {
                    if (notification.id === id) {
                        notification.markAsRead();
                    }

                    return notification;
                });
            },
            [NOTIFICATIONS_MUTATIONS.READ_ALL](state: NotificationsState): void {
                state.notifications = state.notifications.map((notification: UINotification) => {
                    notification.markAsRead();

                    return notification;
                });

                state.unreadCount = 0;
            },
        },
        actions: {
            [NOTIFICATIONS_ACTIONS.GET_NOTIFICATIONS]: async function ({commit}: NotificationsContext, pageIndex: number): Promise<void> {
                const notificationsResponse = await service.notifications(pageIndex);

                const notifications = notificationsResponse.page.notifications.map(notification => new UINotification(notification));

                const notificationState = new NotificationsState(notifications, notificationsResponse.page.pageCount, notificationsResponse.unreadCount);

                commit(NOTIFICATIONS_MUTATIONS.SET_NOTIFICATIONS, notificationState);

                if (pageIndex === 1) {
                    commit(NOTIFICATIONS_MUTATIONS.SET_LATEST, notificationState);
                }
            },
            [NOTIFICATIONS_ACTIONS.MARK_AS_READ]: async function ({commit}: NotificationsContext, id: string): Promise<void> {
                await service.readSingeNotification(id);
                commit(NOTIFICATIONS_MUTATIONS.MARK_AS_READ, id);
            },
            [NOTIFICATIONS_ACTIONS.READ_ALL]: async function ({commit}: NotificationsContext): Promise<void> {
                await service.readAllNotifications();

                commit(NOTIFICATIONS_MUTATIONS.READ_ALL);
            },
        },
    };
}
