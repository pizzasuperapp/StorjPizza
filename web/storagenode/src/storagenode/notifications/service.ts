// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import { NotificationsApi, NotificationsCursor, NotificationsResponse } from '@/storagenode/notifications/notifications';

/**
 * PayoutService is used to store and handle node paystub information.
 * PayoutService exposes a business logic related to payouts.
 */
export class NotificationsService {
    private readonly api: NotificationsApi;

    public constructor(api: NotificationsApi) {
        this.api = api;
    }

    /**
     * Fetch notifications.
     *
     * @returns notifications response.
     * @throws Error
     */
    public async notifications(index: number, limit?: number): Promise<NotificationsResponse> {
        const cursor = new NotificationsCursor(index, limit);

        return await this.api.get(cursor);
    }

    /**
     * Marks single notification as read on server.
     * @param id
     */
    public async readSingeNotification(id: string): Promise<void> {
        await this.api.read(id);
    }

    /**
     * Marks all notifications as read on server.
     */
    public async readAllNotifications(): Promise<void> {
        await this.api.readAll();
    }
}
