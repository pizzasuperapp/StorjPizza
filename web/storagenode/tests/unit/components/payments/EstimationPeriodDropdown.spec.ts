// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vue from 'vue';
import Vuex from 'vuex';

import EstimationPeriodDropdown from '@/app/components/payments/EstimationPeriodDropdown.vue';

import { appStateModule } from '@/app/store/modules/appState';
import { newNodeModule, NODE_MUTATIONS } from '@/app/store/modules/node';
import { StorageNodeApi } from '@/storagenode/api/storagenode';
import { StorageNodeService } from '@/storagenode/sno/service';
import {
    Dashboard,
    Satellite,
    SatelliteInfo, SatelliteScores,
    Stamp,
    Traffic,
} from '@/storagenode/sno/sno';
import { createLocalVue, shallowMount } from '@vue/test-utils';

const nodeApi = new StorageNodeApi();
const nodeService = new StorageNodeService(nodeApi);
const nodeModule = newNodeModule(nodeService);

const localVue = createLocalVue();
localVue.use(Vuex);

Vue.directive('click-outside', {
    bind: (): void => { return; },
    unbind: (): void => { return; },
});

const store = new Vuex.Store({ modules: { appStateModule, node: nodeModule }});

describe('EstimationPeriodDropdown', (): void => {
    it('renders correctly', (): void => {
        const wrapper = shallowMount(EstimationPeriodDropdown, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('opens calendar on click only when historical data exists',   async (): Promise<void> => {
        const satelliteInfo = new Satellite(
            '3',
            [new Stamp()],
            [],
            [],
            [],
            111,
            222,
            50,
            70,
            new SatelliteScores('', 1, 0, 0),
            new Date(),
        );

        const dashboardInfo = new Dashboard(
            '1',
            '2',
            [],
            [
                new SatelliteInfo('3', 'url1', null, null),
                new SatelliteInfo('4', 'url2', new Date(), new Date(2020, 0, 1)),
            ],
            new Traffic(99, 100, 4),
            new Traffic(50),
            new Date(),
            new Date(),
            '0.1.1',
            '0.2.2',
            false,
            true,
            '13000',
        );

        store.commit(NODE_MUTATIONS.POPULATE_STORE, dashboardInfo);
        store.commit(NODE_MUTATIONS.SELECT_SATELLITE, satelliteInfo);

        const wrapper = shallowMount(EstimationPeriodDropdown, {
            store,
            localVue,
        });

        wrapper.find('.period-container').trigger('click');

        await localVue.nextTick();
        expect(wrapper.find('.period-container__calendar').exists()).toBe(false);

        satelliteInfo.joinDate = new Date(Date.UTC(2019, 10, 29));

        store.commit(NODE_MUTATIONS.SELECT_SATELLITE, satelliteInfo);

        wrapper.find('.period-container').trigger('click');

        await localVue.nextTick();
        expect(wrapper.find('.period-container__calendar').exists()).toBe(true);
    });
});
