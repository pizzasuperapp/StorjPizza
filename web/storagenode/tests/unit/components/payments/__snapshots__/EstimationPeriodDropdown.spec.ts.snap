// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EstimationPeriodDropdown renders correctly 1`] = `
<button name="Show Period Date Picker" type="button" class="period-container">
  <p class="period-container__label long-text">Custom Date Range</p>
  <p class="period-container__label short-text">Custom Range</p>
  <blackarrowexpand-stub></blackarrowexpand-stub>
  <!---->
</button>
`;
